### DISCUSSION TITLE
How to provide size variant based on media query without using javascript?

### DISCUSSION THREAD

**Discussion by lorcan-codes** [joe-bell/cva#290] [Q&A]
If we want to provide different size variants (sm, md, lg) depending on window size, we have to track window size changes

```jsx
const isMobile = useMediaQuery("max-width : 768px")

return <Button size={isMobile? "sm": "md"}>My button</Button>
```

But this approach requires using client side component and tracking window size. I wonder if there is any way to make this work in server components and without tracking window size

**Comment by Ganbin** [comment]
Same here, I want a button to be `size="sm"` when I hit the "sm" Tailwind mediaQuery. Without using javascript if possible.
